test/TestServerProjectsLowLevel/dpg-customization/Models/Input.Serialization.cs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Text.Json;
+using Azure.Core;
+
+namespace dpg_customization_LowLevel.Models
+{
+    public partial class Input : IUtf8JsonSerializable
```

Comment:
I need to double check if we are ok using `IUtf8JsonSerializable` for these tests.

Reply:
The IUtf8JsonSerializable is generated from HLC code and I use this to serilization from model to requestContent. Could you let me know the concrete points you want to check? Performance? And do we need to review it again in HLC?

Reply:
[IUtf8JsonSerializable ](https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Generator.Shared/IUtf8JsonSerializable.cs) is an internal type, so I'm not sure what's happening there.  If the model is in the public API, it will not be able to inherit from an internal type.

Reply:
Do we still need to consider the internal type since we only consider #4 now? I think the serialization part is the internal process now if I understand correctly.

Reply:
Yes, serialization should all happen internally, so that it is not visible in the public API.

Some examples of public HLC models are:
- [ArtifactManifestProperties](https://apiview.dev/Assemblies/Review/7ebfcb36ff6149ba96170bd059094513#Azure.Containers.ContainerRegistry.ArtifactManifestProperties)
- [KeyVaultKey](https://apiview.dev/Assemblies/Review/d2d05cf175c7478184662d580a300e68#Azure.Security.KeyVault.Keys.KeyVaultKey)

Reply:
Sorry I didn't get. Do you mean we shouldn't use IUtf8JsonSerializable because it is internal? But we still could create Input outside the assembly because Input is public.

Reply:
I see from this APIView that the serialization methods are not in the public API: https://apiview.dev/Assemblies/Review/6dc0d77581714f0f83cf47a95b3c10cb#dpg_customization_LowLevel.Models.Input

You may disregard my comment.